### ISSUE TITLE
Atom/RSS feeds URL

### ISSUE THREAD

**Issue by BoboTiG** [BoboTiG/blogotext#153]
Pour afficher un flux, nous passerons par `index.php?mode=atom|rss` (`mode` est utilisé pour les commentaires et liens, comme ça on reste dans la logique sans ajouter une nouvelle variable `$ GET`).

Du coup, suppression des fichiers `atom.php`,  `feed.php` et  `rss.php`.

**Comment by remrem** [comment]
Je mets à jours les fichiers des feed pour passer à `?mode=`
Je mets aussi à jour le théme par défault

**Comment by remrem** [comment]
`index.php?mode=atom|rss` ne peut pas fonctionner, déjà utiliser pour utiliser les feed de blog, comments ...

**Comment by BoboTiG** [comment]
Nous aviserons après la 3.7, mais sur le principe, ça te convient ?

**Comment by remrem** [comment]
de passer le format de sortie dans `?mode=` ?

**Comment by BoboTiG** [comment]
Yep, on avait dit que, pour la transition, on utiliserait `feed.php` mais je me rends compte de ma bétise. 
On peut tout de même ajouter `rss` et `atom` dans les modes, non ? Le code de `index.php` :

    // paramètre mode : quelle table "mode" ?
    if (isset($_GET['mode'])) {
        switch ($_GET['mode']) {
            case 'comments':
                $query = "SELECT c.*, a.bt_title FROM ";
                $where = 'commentaires';
                $GLOBALS['tpl_class'] .= 'content-comments ';
                break;
            case 'links':
                $where = 'links';
                $GLOBALS['tpl_class'] .= 'content-links ';
                break;
            case 'blog':
            default:
                $where = 'articles';
                $GLOBALS['tpl_class'] .= 'content-blog ';
                break;
        }

Dans ce `switch`, les deux nouvelles possibilités iraient bien je pense.

**Comment by remrem** [comment]
Le problème est, comment distingué proprement le flux des links au format atom du flux des links au format rss ?. via ?mode= à part à utiliser des ?mode=atom-links ?mode=rss-links, je ne vois pas trop comment faire :s

**Comment by remrem** [comment]
Ce qu'il faut se dire, c'est que là, on a des "pseudo alias" rss.php & atom.php -> feed.php, ça fonctionne, l'utilisateur ne le voit/sent pas. Certes ça fait 1 include pour pas grand chose, mais ce n'est pas dramatique avec ce qu'on a optimisé à côté. On peux laisser dans l'état histoire de simplifier la maintenance et les MAJ des process des script des flux atom et php, et le temps de se décider, même si on vire rss.php & atom.php à la 4.99 ou 5.1, ce n'est pas dramatique non plus ...

**Comment by BoboTiG** [comment]
Bonne idée, on freeze pour la 3.7 avec `feed.php` et réfléchira à comment gérer ça pour la v4. Mais pour rester logique : `mode=blog|comment|link&format=atom|rss` serait pas mal. À voir plus tard, j'ai épuisé mon quota de temps de réponse pour ajourd'hui ;)
